addon/webextension/sitehelper.js

```diff
@@ -0,0 +1,41 @@
+/* globals catcher, callBackground */
+/** This is a content script added to all Page Shot pages, and allows the site to
+    communicate with the add-on */
+
+window.sitehelper = (function () {
+
+  catcher.registerHandler((errorObj) => {
+    callBackground("reportError", errorObj);
+  });
+
+
+  function sendCustomEvent(name, detail) {
+    if (typeof detail == "object") {
```

Comment:
I guess this is okay: it's not an accurate check for objecthood, but it's unlikely that `null`, an array, or some oddball object like `Math` would ever get passed in here.